### PR TITLE
Re-enable doctests for the quicktour

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR re-enables the doctests for the quicktour. It shouldn't be merged without its [companion PR](https://github.com/huggingface/doc-builder/pull/112) on the doc-builder side to properly recognize the new framework split comment and to remove the doctest flags on some lines.

I've adapted the example applying the pipeline on the whole superb dataset because:
1. it was way too long
2. writing a custom generator using an object not documented and not even in the main init of Transformers is too advanced for the quicktour.

I've also update the guide for doctests.

### CODE REVIEW

File: docs/source/quicktour.mdx [hunk 1]
```diff
@@ -112,20 +112,22 @@ Next, load a dataset (see the 🤗 Datasets [Quick Start](https://huggingface.co
...
 
 ```py
->>> from transformers.pipelines.pt_utils import KeyDataset
->>> from tqdm.auto import tqdm
-
->>> for out in tqdm(speech_recognizer(KeyDataset(dataset, "file"))):
-...     print(out)
-{"text": "HE HOPED THERE WOULD BE STEW FOR DINNER TURNIPS AND CARROTS AND BRUISED POTATOES AND FAT MUTTON PIECES TO BE LADLED OUT IN THICK PEPPERED FLOWER FAT AND SAUCE"}
+>>> files = dataset["file"]
+>>> speech_recognizer(files[:4])
```
Comment (sgugger) [inline docs/source/quicktour.mdx:122]: I think it's better to just load a small portion of the dataset here for something that executes quickly and is easier to understand than using `KeyDataset`. cc @stevhliu

File: docs/source/quicktour.mdx [hunk 2]
```diff
@@ -141,7 +143,7 @@ Use the [`AutoModelForSequenceClassification`] and ['AutoTokenizer'] to load the
 
 >>> model = AutoModelForSequenceClassification.from_pretrained(model_name)
 >>> tokenizer = AutoTokenizer.from_pretrained(model_name)
-===PT-TF-SPLIT===
+>>> # ===PT-TF-SPLIT===
```
Comment (patrickvonplaten) [inline docs/source/quicktour.mdx:146]: Nice!

File: conftest.py [hunk 3]
```diff
@@ -59,3 +60,17 @@ def pytest_sessionfinish(session, exitstatus):
     # If no tests are collected, pytest exists with code 5, which makes the CI fail.
     if exitstatus == 5:
         session.exitstatus = 0
+
+
+# Doctest custom flag to ignore output.
+IGNORE_RESULT = doctest.register_optionflag('IGNORE_RESULT')
```
Comment (patrickvonplaten) [inline conftest.py:66]: Cool! Very useful :-)

**Review by patrickvonplaten — APPROVED** [review]
Thanks a lot!

File: conftest.py (hunk 3 above)
Comment (LysandreJik) [reply conftest.py:66]: Very cool, was unaware of that flag.

File: docs/README.md [hunk 4]
```diff
@@ -340,14 +340,47 @@ seen [here](https://github.com/huggingface/transformers/actions/workflows/doctes
...
+  * whitespace: one give whitespace (space, tabulation, new line) is equivalent to any number of whitespace, so you can add new lines where there are spaces to make your output more readable.
+  * numerical values: you should never put more than 4 or 5 digits to expected results as different setups or library versions might get you slightly different results. `doctest` is configure to ignore any difference lower than the precision to which you wrote (so 1e-4 if you write 4 digits).
+- Don't leave a block of code that is very long to execute. If you can't make it fast, you can either not use the doctest syntax on it (so that it's ignored), or if you want to use the doctest syntax to show the results, you can add a comment `# doctest: +SKIP` at the end of the lines of code too long to execute
+- Each line of code that produces a result needs to have that result written below. You can ignore an output if you don't want to show it in your code example by adding a comment ` # doctest: +IGNORE_RESULT` at the end of the line of code produing it.
```
Comment (LysandreJik) [inline docs/README.md:386]: Thanks for completing that, super useful

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @sgugger!

**Pull request merged**